mod/node-core/pkg/components/dispatcher.go

```diff
@@ -23,29 +23,106 @@ package components
 import (
 	"cosmossdk.io/depinject"
 	"github.com/berachain/beacon-kit/mod/async/pkg/dispatcher"
-	asynctypes "github.com/berachain/beacon-kit/mod/async/pkg/types"
 	"github.com/berachain/beacon-kit/mod/log"
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/async"
 )
 
 // DispatcherInput is the input for the Dispatcher.
 type DispatcherInput[
 	LoggerT any,
 ] struct {
 	depinject.In
-	Logger     LoggerT
-	Publishers []asynctypes.Broker
+	Logger LoggerT
 }
 
 // ProvideDispatcher provides a new Dispatcher.
 func ProvideDispatcher[
+	BeaconBlockT any,
+	BlobSidecarsT any,
 	LoggerT log.AdvancedLogger[any, LoggerT],
 ](
 	in DispatcherInput[LoggerT],
 ) (*Dispatcher, error) {
+	var err error
 	d := dispatcher.New(
 		in.Logger.With("service", "dispatcher"),
```

Comment:
WithEvent(....) dispatcher.Option() thingy (lots of examples in the repo)

Reply:
yessirrr